tensorboard/plugins/hparams/tf_hparams_parallel_coords_plot/axes.ts

```diff
@@ -0,0 +1,554 @@
+   * The actual position of an axis is either its associated stationary 
+   * position if its not dragged or its currently dragged position. This class
+   * maintains the invariant that the axes' stationary positions match the order
+   * of their actual position by re-assigning stationary positions to axes when
+   * dragging an axis causes it to "pass" another axes.
+   */
+  public dragStart(colIndex: number) {
+    console.assert(!this.isAxisDragging());
+    console.assert(this._axes[colIndex].isDisplayed());
+    this._draggedAxis = this._axes[colIndex];
+    this._draggedAxisPosition = this._stationaryAxesPositions(colIndex);
+  }
+
+  public drag(newX: number) {
+    newX = Math.min(Math.max(newX, 0), this._svgProps.width);
+    this._draggedAxisPosition = newX;
+    let visibleColIndices = this._stationaryAxesPositions.domain();
+    visibleColIndices.sort(
+      (ci1, ci2) => this.getAxisPosition(ci1) - this.getAxisPosition(ci2));
+    this._stationaryAxesPositions.domain(visibleColIndices);
+    this._updateAxesPositionsInDOM(this._parentsSel);
+  }
+
+  public dragEnd(duration: number) {
+    console.assert(this.isAxisDragging());
+    this._draggedAxisPosition = null;
+    this._draggedAxis = null;
+    this._updateAxesPositionsInDOM(
+      this._parentsSel.transition().duration(duration))
+  }
+
+  public isAxisDragging(): boolean {
+    return this._draggedAxis !== null;
+  }
+
+  public getAxisPosition(colIndex: number) : number {
+    return (this._draggedAxis !== null) &&
+      (this._draggedAxis.colIndex() === colIndex)
+      ? this._draggedAxisPosition
+      : this._stationaryAxesPositions(colIndex);
+  }
+
+  /**
+   * Sets the domain of 'stationaryAxesPositions' to be precisely the given 
+   * visibleColIndices, but preserves the order of the column indices that
+   * are already in the domain. Essentially, this method removes indices in 
+   * the domain that are not in visibleColIndices and then appends indices in 
+   * visibleColIndices that are not currently in the domain. Thus, indices in 
+   * visibleColIndices that are already in stationaryAxesPositions
+   * will maintain their order in stationaryAxesPositions and will precede
+   * the new elements.
+   *
+   * This reassigns stationary positions to axes so that the only visible 
+   * axes are the ones with column indices in 'visibleColIndices', but preserves
+   * the order of axes indexed by visibleColIndices that are already visible.
+   */
+  private _updateStationaryAxesPositions(visibleColIndices: Set<number>) {
+    // We're going to modify visibleColIndices so make a copy first, since
+    // the caller may count on it being unmodified.
+    visibleColIndices = new Set<number>(visibleColIndices);    
+    let newDomain: number[] = this._stationaryAxesPositions.domain().filter(
+      colIndex => visibleColIndices.has(colIndex));
+    newDomain.forEach(colIndex => visibleColIndices.delete(colIndex));
```

Comment:
I don't think comments are panacea and making code a little bit readable will help. Please take a look at code below and tell me what you think.
```ts
// Sort visibleColIndices so indicies present in _stationaryAxesPositions
// appear before others.
const visibleDomain = this._stationaryAxesPositions.domain()
     .filter(colIndex => visibleColIndices.has(colIndex));
const newDomain  = Array.from(new Set([
    ...Array.from(visibleDomain),
    ...Array.from(visibleColIndices),
]);
this._stationaryAxesPositions.domain(newDomain);
 ```